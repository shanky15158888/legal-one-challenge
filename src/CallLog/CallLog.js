import React from 'react'
import Table from '../Table/Table';
import logs from '../json-data/logs.json'
import resolution from '../json-data/resolution.json'
import agents from '../json-data/agents.json'
import moment from 'moment';
import { Link } from 'react-router-dom';

export default function CallLog(props) {
    const number = props.match && props.match.params && props.match.params.number ? props.match.params.number : "";
    const cols = ["Agent Name", "Call date and time", "Resolution"]
    const tableData = [];

    const recordsCalledAtNumber = logs.filter((log) => {
        return log.number === number
    })

    const agentNames = getAgentNames(recordsCalledAtNumber)

    const resolutions = getResolution(agentNames);

    resolutions.forEach((item) => {
        const obj = {}
        obj['data'] = []
        obj.data.push({ 'data': item.name })
        obj.data.push({ 'data': getDateTimeFormat(item.dateTime) })
        obj.data.push({ 'data': item.resolution })
        tableData.push(obj)
    })

    return (
        <>
            <header><h2>Call Logs</h2></header>
            <Link className="goRight" to="/">Home</Link>
            <Table cols={cols} rows={tableData} />
        </>
    )
}

const getAgentNames = (recordsCalledAtNumber) => {
    const arr = JSON.parse(JSON.stringify(recordsCalledAtNumber));
    const temp_arr = [];
    arr.forEach((item) => {
        agents.forEach((subItem) => {
            if (item.agentIdentifier === subItem.identifier) {
                item['name'] = `${subItem.firstName} ${subItem.lastName}`
                temp_arr.push(item)
            }
        })
    })
    return temp_arr
}

const getResolution = (agentNameList) => {
    const arr = JSON.parse(JSON.stringify(agentNameList));
    const temp_arr = [];
    arr.forEach((item) => {
        resolution.forEach((subItem) => {
            if (item.identifier === subItem.identifier) {
                item['resolution'] = subItem.resolution
                temp_arr.push(item)
            }
        })
    })
    return temp_arr
}

const getDateTimeFormat = (date) => {
    return moment(date).format('DD/M/YYYY HH:MM:SS');
}
