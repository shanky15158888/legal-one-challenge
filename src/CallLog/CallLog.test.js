import React from 'react';
import renderer from 'react-test-renderer';
import CallLog from './CallLog';
import { BrowserRouter as Router } from "react-router-dom";

test('CallLog component testing', () => {
    const match = {
        params: {
            number: "+49151484522"
        }
    }
    const component = renderer.create(
        <Router>
            <CallLog match={match} />
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});