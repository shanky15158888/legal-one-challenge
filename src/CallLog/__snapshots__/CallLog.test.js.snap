// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallLog component testing 1`] = `
Array [
  <header>
    <h2>
      Call Logs
    </h2>
  </header>,
  <a
    className="goRight"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>,
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Agent Name
        </th>
        <th
          scope="col"
        >
          Call date and time
        </th>
        <th
          scope="col"
        >
          Resolution
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          05/10/2020 20:10:00
        </td>
        <td>
          need reschedule
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          06/10/2020 19:10:00
        </td>
        <td>
          interested
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          interested
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          no answer
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          no answer
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          interested
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          needs follow up
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          needs follow up
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          needs follow up
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          no answer
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          interested
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          interested
        </td>
      </tr>
      <tr
        scope="col"
      >
        <td>
          Abraham Ellis
        </td>
        <td>
          07/10/2020 20:10:00
        </td>
        <td>
          needs follow up
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
